traits/ctraits.c

```diff
@@ -4851,6 +4851,7 @@ set_trait_post_setattr(trait_object *trait, PyObject *value, void *closure)
     if (value == Py_None) {
         value = NULL;
         trait->post_setattr = NULL;
+        Py_DECREF(Py_None);
```

Comment:
This isn't right: `set_trait_post_setattr` doesn't own a reference to `Py_None` in this case, so you don't want to decref it. The CPython reference ownership rules in general are messy, but they're summarised here: https://docs.python.org/3/extending/extending.html#ownership-rules

The relevant part at this point is this part:

> When you pass an object reference into another function, in general, the function borrows the reference from you

If you try running something like this loop on this branch:

```
    for _ in range(100):
        some_trait.post_setattr = some_callable
        some_trait.post_setattr = None
        print(sys.getrefcount(None))
```

you'll see the reference count of `None` decreasing by one on every iteration (and if you increase the loop count to something big enough, like `10000` or more, you'll see a segmentation fault).

Reply:
Thanks for the link and the code for making sure! Reverted the change.